File: 1833-find-the-highest-altitude/find-the-highest-altitude.go
func largestAltitude(gain []int) int {
    var res int
    if 0>gain[0]{
        res=0
    }else{
        res=gain[0]
    }
    for i:=1;i<len(gain);i++{
        gain[i]+=gain[i-1]
        if res<gain[i]{
            res=gain[i]
        }
    }
    return res
    
}

File: 1833-find-the-highest-altitude/find-the-highest-altitude.java
class Solution {
    public int largestAltitude(int[] gain) {
        ArrayList<Integer> res=new ArrayList<>();
        res.add(0);
        int total=0;
        for(int i=0;i<gain.length;i++){
            total+=gain[i];
            res.add(total);
        }
        return Collections.max(res);
    }
}